### PR TITLE
Add load_earth_free_air_anomaly function for Earth free-air anomaly dataset

### PR DESCRIPTION
**Pull request by willschlitzer**
This adds `load_earth_free_air_anomaly` to load the IGPP Global Earth Free-Air Anomaly dataset.

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/datasets/load_remote_dataset.py [hunk 1]
```diff
@@ -147,6 +147,26 @@ class GMTRemoteDataset(NamedTuple):
...
+            "30m": Resolution(["pixel", "gridline"], False),
+            "20m": Resolution(["pixel", "gridline"], False),
+            "15m": Resolution(["pixel", "gridline"], False),
+            "10m": Resolution(["pixel", "gridline"], False),
+            "06m": Resolution(["pixel", "gridline"], False),
+            "05m": Resolution(["pixel", "gridline"], True),
+            "04m": Resolution(["pixel", "gridline"], True),
+            "03m": Resolution(["pixel", "gridline"], True),
+            "02m": Resolution(["pixel", "gridline"], True),
+            "01m": Resolution(["gridline"], True),
```
Comment (seisman) [inline pygmt/datasets/load_remote_dataset.py:167]: `01m` grid is pixel-registered.
https://www.generic-mapping-tools.org/remote-datasets/earth-faa.html
```suggestion
            "01m": Resolution(["pixel"], True),
```

File: pygmt/tests/test_datasets_earth_free_air_anomaly.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            load_earth_free_air_anomaly(resolution=resolution)
+
+
+def test_earth_faa_incorrect_registration():
+    """
+    Test loading earth_free_air_anomaly with incorrect registration type.
+    """
+    with pytest.raises(GMTInvalidInput):
+        load_earth_free_air_anomaly(registration="improper_type")
+
```
Comment (seisman) [inline pygmt/tests/test_datasets_earth_free_air_anomaly.py:28]: These tests are common to all Earth datasets. Perhaps we can move all these tests in a new test file like `test_datasets_load_remote_dataset.py` instead.

File: pygmt/datasets/earth_free_air_anomaly.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    Note
+    ----
+    The :class:`xarray.DataArray` grid doesn't support slice operation, for
+    Earth free-air anomaly with resolutions of 5 arc-minutes or higher,
+    which are stored as smaller tiles.
+    """
+    dataset_prefix = "earth_faa_"
+    grid = _load_remote_dataset(
+        dataset_name="earth_free_air_anomaly",
+        dataset_prefix=dataset_prefix,
```
Comment (seisman) [inline pygmt/datasets/earth_free_air_anomaly.py:64]:
```suggestion
    grid = _load_remote_dataset(
        dataset_name="earth_free_air_anomaly",
        dataset_prefix="earth_faa_",
```

File: pygmt/helpers/testing.py [hunk 4]
```diff
@@ -186,6 +186,9 @@ def download_test_data():
         # Earth magnetic anomaly grids
         "@earth_mag_01d_g.grd",
         "@S90W180.earth_mag_05m_g.nc",  # Specific grid for 05m test
+        # Earth free-air anomaly grids
+        "@earth_faa_01d_g.grd",
```
Comment (seisman) [inline pygmt/helpers/testing.py:190]:
```suggestion
        "@earth_faa_01d_g",
```

File: pygmt/helpers/testing.py [hunk 5]
```diff
@@ -186,6 +186,9 @@ def download_test_data():
         # Earth magnetic anomaly grids
         "@earth_mag_01d_g.grd",
```
Comment (seisman) [inline pygmt/helpers/testing.py:187]:
```suggestion
        "@earth_mag_01d_g",
```

File: pygmt/helpers/testing.py [hunk 6]
```diff
@@ -186,6 +186,9 @@ def download_test_data():
         # Earth magnetic anomaly grids
```
Comment (seisman) [inline pygmt/helpers/testing.py:186]: Should also remove the `.grd` suffix at line 184.

File: pygmt/tests/test_datasets_earth_free_air_anomaly.py (hunk 2 above)
Comment (seisman) [reply pygmt/tests/test_datasets_earth_free_air_anomaly.py:28]: This can be done in a separate PR if necessary.

**Comment by seisman** [comment]
@willschlitzer This PR looks good to me, but need to resolve the conflicts first.

File: pygmt/datasets/earth_free_air_anomaly.py [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+        The grid resolution. The suffix ``d`` and ``m`` stand for
+        arc-degree and arc-minute. It can be ``"01d"``, ``"30m"``,
+        ``"20m"``, ``"15m"``, ``"10m"``, ``"06m"``, ``"05m"``, ``"04m"``,
+        ``"03m"``, ``"02m"``, or ``"01m"``.
+
+    region : str or list
+        The subregion of the grid to load, in the forms of a list
+        [*xmin*, *xmax*, *ymin*, *ymax*] or a string *xmin/xmax/ymin/ymax*.
+        Required for grids with resolutions higher than 5
+        arc-minute (i.e., ``"05m"``).
```
Comment (michaelgrund) [inline pygmt/datasets/earth_free_air_anomaly.py:41]:
```suggestion
        arc-minutes (i.e., ``"05m"``).
```

**Review by michaelgrund — APPROVED** [review]

File: pygmt/datasets/earth_free_air_anomaly.py (hunk 7 above)
Comment (yvonnefroehlich) [reply pygmt/datasets/earth_free_air_anomaly.py:41]: Looks like there is some inconsistency regarding using singular or plural and this has to be changed also for the other datasets, e.g. [load_earth_age](https://www.pygmt.org/dev/api/generated/pygmt.datasets.load_earth_age.html?highlight=arc-minute), [load_earth_geoid](https://www.pygmt.org/dev/api/generated/pygmt.datasets.load_earth_geoid.html?highlight=arc-minute), etc. I can submit a separate PR, if we agree that plural is better as suggested by @michaelgrund.

**Pull request merged**